pykafka/balancedconsumer.py

```diff
@@ -259,16 +260,19 @@ def stop(self):
         if self._consumer is not None:
             self._consumer.stop()
         self._zookeeper.remove_listener(self._zk_state_listener)
-        if self._owns_zookeeper:
-            # NB this should always come last, so we do not hand over control
-            # of our partitions until consumption has really been halted
-            self._zookeeper.stop()
-        else:
-            self._remove_partitions(self._get_held_partitions())
-            self._zookeeper.delete(self._path_self)
-            # additionally we'd want to remove watches here, but there are no
-            # facilities for that in ChildrenWatch - as a workaround we check
-            # self._running in the watcher callbacks (see further down)
+        try:
+            if self._owns_zookeeper:
+                # NB this should always come last, so we do not hand over control
+                # of our partitions until consumption has really been halted
+                self._zookeeper.stop()
+            else:
+                self._remove_partitions(self._get_held_partitions())
+                self._zookeeper.delete(self._path_self)
+                # additionally we'd want to remove watches here, but there are no
+                # facilities for that in ChildrenWatch - as a workaround we check
+                # self._running in the watcher callbacks (see further down)
+        except NoNodeError:
+            log.warning("Error encountered while closing ZooKeeper connection")
```

Comment:
Minor nitpick: this makes it sound as if something unexpected happened, when in fact we know why this happens. If we'd wrap the `try .. except` around the `delete(_path_self)` line only, I'd argue the error would be so specific that you could just `pass` without logging it?